lib/pacio_test_kit/interactions_test.rb

```diff
@@ -1,6 +1,28 @@
 module PacioTestKit
   module InteractionsTest
     # TODO: All helper methods for interactions tests (CRUD) will be added here.
+
+    def update_and_validate_resource(retrieved_request)
+      retrieved_fhir_resource = FHIR.from_contents(JSON.parse(retrieved_request.response_body).to_json)
+
+      modified_fhir_resource = modify_resource_status(retrieved_fhir_resource.deep_dup)
+
+      fhir_update(modified_fhir_resource, retrieved_fhir_resource.id)
+      assert_response_status(200)
+
+      validate_resource_status(retrieved_fhir_resource, resource)
```

Comment:
I don't think we need extract the logic for modifying the resource and validating the status here. These should be one line code each. e.g. can directly do `retrieved_fhir_resource.status = 'registered'` .  even if you assign it to a new variable, it's still the same object. That's why this test will always fail because the retrieved_resource is modified in modify_resource_status, which means the status of the resource is no longer the original one when it is passed into the assert in validate_resource_status. the status will be the same as the resource returned from the update request.

Also, we should not have a fixed status to update to, since we do not know what the incoming status is.